hrjobcontract/CRM/Hrjobcontract/Upgrader.php

```diff
@@ -1252,6 +1255,48 @@ public function upgrade_1036() {
     return TRUE;
   }
 
+  /**
+   * Create new option group hrleaveandabsences_work_pattern_change_reason
+   * with values As per contract, Change in contractual hours and
+   * Change in contract type.
+   *
+   * @return bool
+   */
+  public function upgrade_1037() {
+    $result = civicrm_api3('OptionGroup', 'get', [
```

Comment:
This option group will be used by leave and absences, so it should be added there, not here. Besides that, new option groups should be defined using a XML file, like the ones we have [here](https://github.com/civicrm/civihr/tree/staging/uk.co.compucorp.civicrm.hrleaveandabsences/xml/option_groups). This XML will be loaded automatically when creating a new site, but for existing site you'll still need an upgrader. Check if it's possible to make the upgrader use the definition from the XML file.

Finally, we might want to make this Option Group reserved. I don't think it will be a good idea to let the users delete it (I think it's ok if they update or change the option values though).

Reply:
This will be migrated as instructed. A new ticket has been created to handle defining new option group in the XML file like the ones suggested.